conformance/05_referrers.go

```diff
@@ -0,0 +1,276 @@
+				req := client.NewRequest(reggie.POST, "/v2/<name>/blobs/uploads/")
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				req = client.NewRequest(reggie.PUT, resp.GetRelativeLocation()).
+					SetQueryParam("digest", testRefBlobADigest).
+					SetHeader("Content-Type", "application/octet-stream").
+					SetHeader("Content-Length", testRefBlobALength).
+					SetBody(testRefBlobA)
+				resp, err = client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(SatisfyAll(
+					BeNumerically(">=", 200),
+					BeNumerically("<", 300)))
+			})
+
+			g.Specify("Populate registry with test references manifest (config.MediaType = artifactType)", func() {
+				SkipIfDisabled(referrers)
+				RunOnlyIf(runReferencesSetup)
+				req := client.NewRequest(reggie.PUT, "/v2/<name>/manifests/<reference>",
+					reggie.WithReference(refsManifestConfigArtifactDigest)).
+					SetHeader("Content-Type", "application/vnd.oci.image.manifest.v1+json").
+					SetBody(refsManifestConfigArtifactContent)
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(SatisfyAll(
+					BeNumerically(">=", 200),
+					BeNumerically("<", 300)))
+			})
+
+			g.Specify("Populate registry with test references manifest (ArtifactType, config.MediaType = scratch)", func() {
+				SkipIfDisabled(referrers)
+				RunOnlyIf(runReferencesSetup)
+				req := client.NewRequest(reggie.PUT, "/v2/<name>/manifests/<reference>",
+					reggie.WithReference(refsManifestLayerArtifactDigest)).
+					SetHeader("Content-Type", "application/vnd.oci.image.manifest.v1+json").
+					SetBody(refsManifestLayerArtifactContent)
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(SatisfyAll(
+					BeNumerically(">=", 200),
+					BeNumerically("<", 300)))
+			})
+		})
+
+		g.Context("Get references", func() {
+			g.Specify("GET request to nonexistent blob should result in empty 200 response", func() {
+				SkipIfDisabled(referrers)
+				req := client.NewRequest(reggie.GET, "/v2/<name>/referrers/<digest>",
+					reggie.WithDigest(dummyDigest))
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(Equal(http.StatusOK))
+
+				var index Index
+				err = json.Unmarshal(resp.Body(), &index)
+				Expect(err).To(BeNil())
+				Expect(len(index.Manifests)).To(BeZero())
+			})
+
+			g.Specify("GET request to existing blob should yield 200", func() {
+				SkipIfDisabled(referrers)
+				req := client.NewRequest(reggie.GET, "/v2/<name>/referrers/<digest>",
+					reggie.WithDigest(manifests[4].Digest))
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(Equal(http.StatusOK))
+				if h := resp.Header().Get("Docker-Content-Digest"); h != "" {
+					Expect(h).To(Equal(configs[4].Digest))
+				}
+
+				var index Index
+				err = json.Unmarshal(resp.Body(), &index)
+				Expect(err).To(BeNil())
+				Expect(len(index.Manifests)).ToNot(BeZero())
```

Comment:
We should test for length >= 2, and check for the existence of the two digests in the manifest list.

Reply:
Fixed.